books/books/spiders/books.py

```diff
@@ -0,0 +1,55 @@
+from typing import Generator
+
+import scrapy
+from scrapy.http import Response, Request
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
```

Comment:
Pls check the officials docs and follow the recommendations :

https://docs.scrapy.org/en/latest/intro/tutorial.html